### PR TITLE
Enforce prow postsubmit/periodic job has a match testgrid tab

### PR DESCRIPTION
**Pull request by krzyzacy**
https://github.com/kubernetes/test-infra/issues/1654

Add some protection before we move stuff over.

### CODE REVIEW

File: prow/config/jobs.go [hunk 1]
```diff
@@ -150,34 +150,51 @@ func (c *Config) SetPresubmits(jobs map[string][]Presubmit) error {
...
 
-func (c *Config) AllJobNames() []string {
-	var listPres func(ps []Presubmit) []string
-	var listPost func(ps []Postsubmit) []string
-	listPres = func(ps []Presubmit) []string {
-		res := []string{}
-		for _, p := range ps {
-			res = append(res, p.Name)
-			res = append(res, listPres(p.RunAfterSuccess)...)
+func (c *Config) ListJobNames(presubmit, postsubmit, periodic bool) []string {
```
Comment (spxtr) [inline prow/config/jobs.go]: Make three separate functions.

Comment (krzyzacy) [inline prow/config/jobs.go]: done

File: prow/config/jobs.go [hunk 2]
```diff
@@ -150,9 +150,9 @@ func (c *Config) SetPresubmits(jobs map[string][]Presubmit) error {
 	return nil
 }
 
-func (c *Config) AllJobNames() []string {
+func (c *Config) ListPresubmit() []string {
```
Comment (spxtr) [inline prow/config/jobs.go]: Call this `AllPresubmits` and so on.

File: prow/config/jobs.go [hunk 3]
```diff
@@ -150,9 +150,9 @@ func (c *Config) SetPresubmits(jobs map[string][]Presubmit) error {
 	return nil
 }
 
-func (c *Config) AllJobNames() []string {
+func (c *Config) AllPresubmit() []string {
```
Comment (spxtr) [inline prow/config/jobs.go]: Almost, but it should be `AllPresubmits` since it returns a list of `Presubmit`.

**Pull request merged**